crates/import/src/rest.rs

```diff
@@ -0,0 +1,664 @@
+//! files. VSCode: https://github.com/Huachao/vscode-restclient
+//! Jetbrains: https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html
+
+use anyhow::anyhow;
+use indexmap::IndexMap;
+use itertools::Itertools;
+use serde::de::IgnoredAny;
+use slumber_core::{
+    collection::{
+        Authentication, Chain, ChainId, ChainOutputTrim, ChainSource,
+        Collection, HasId, Method, Profile, ProfileId, Recipe, RecipeBody,
+        RecipeId, RecipeNode, RecipeTree, SelectorMode,
+    },
+    http::content_type::ContentType,
+    template::{Identifier, Template},
+};
+
+use mime::APPLICATION_JSON;
+use reqwest::header::CONTENT_TYPE;
+use rest_parser::{
+    headers::Authorization as RestAuthorization,
+    template::{Template as RestTemplate, TemplatePart as RestTemplatePart},
+    Body as RestBody, RestFlavor, RestFormat, RestRequest, RestVariables,
+};
+use std::path::Path;
+use tracing::error;
+
+/// Convert a VSCode `.rest` file or a Jetbrains `.http` file into a slumber
+/// collection
+pub fn from_rest(rest_file: impl AsRef<Path>) -> anyhow::Result<Collection> {
+    let rest_file = rest_file.as_ref();
+    // Parse the file and determine the flavor using the extension
+    let rest_format = RestFormat::parse_file(rest_file)?;
+    let collection = try_build_collection(rest_format)?;
+    Ok(collection)
+}
+
+/// In Rest "Chains" and "Requests" are connected
+/// The only chain is loading from a file
+#[derive(Debug)]
+struct CompleteRecipe {
+    recipe: Recipe,
+    chain: Option<Chain>,
+}
+
+#[derive(Debug)]
+struct CompleteBody {
+    recipe_body: RecipeBody,
+    chain: Option<Chain>,
+}
+
+/// Convert a REST Template into a Slumber Template
+fn try_build_slumber_template(
+    template: RestTemplate,
+) -> anyhow::Result<Template> {
+    // Rest templates allow spaces in variables
+    // For example `{{ HOST}}` or `{{ HOST }}`
+    // These must be removed before putting it through the slumber
+    // template parser
+    let raw_template = template
+        .parts
+        .into_iter()
+        .map(|part| match part {
+            RestTemplatePart::Text(text) => text,
+            RestTemplatePart::Variable(var) => {
+                "{{".to_string() + var.as_str() + "}}"
+            }
+        })
+        .join("");
+
+    raw_template
+        .parse()
+        .map_err(|err| anyhow!("Failed to parse REST template! {err}"))
+}
+
+/// Convert a map of REST templates to Slumber templates (like headers or
+/// queries)
+/// Errors will be logged and skipped if an invalid template is passed in
+fn build_slumber_templates(
+    template_map: IndexMap<String, RestTemplate>,
+) -> IndexMap<String, Template> {
+    template_map
+        .into_iter()
+        .filter_map(|(k, v)| match try_build_slumber_template(v) {
+            Ok(t) => Some((k, t)),
+            Err(err) => {
+                error!("{err}");
+                None
+            }
+        })
```

Comment:
There's a `ResultTraced` extension in `slumber_core` that makes this simpler:
```suggestion
        .filter_map(|(k, v)| try_build_slumber_template(v).map(|v| (k,v)).traced().ok())
```
Same goes for the other 2 spots with this pattern